src/tool/GenerateAction.cpp

```diff
@@ -13,15 +13,108 @@
 #include "lib/Lib/AbsoluteCompilationDatabase.hpp"
 #include "lib/Lib/ConfigImpl.hpp"
 #include "lib/Lib/CorpusImpl.hpp"
+#include "llvm/Support/Program.h"
 #include <mrdocs/Generators.hpp>
 #include <mrdocs/Support/Error.hpp>
 #include <mrdocs/Support/Path.hpp>
 #include <clang/Tooling/JSONCompilationDatabase.h>
+
 #include <cstdlib>
+#include <unordered_map>
 
 namespace clang {
 namespace mrdocs {
 
+std::optional<std::string> 
+getCompilerInfo(llvm::StringRef compiler) 
+{
+    llvm::SmallString<128> outputPath;
+    if (auto EC = llvm::sys::fs::createTemporaryFile("compiler-info", "txt", outputPath)) 
+    {
+        return std::nullopt;
+    }
+
+    std::optional<llvm::StringRef> redirects[] = {llvm::StringRef(), llvm::StringRef(), outputPath.str()};
+    std::vector<llvm::StringRef> args = {compiler, "-v", "-E", "-x", "c++", "-"};
+
+    llvm::ErrorOr<std::string> compilerPath = llvm::sys::findProgramByName(compiler);
+    if ( ! compilerPath) 
+    {
+        return std::nullopt;
+    }
+
+    int result = llvm::sys::ExecuteAndWait(*compilerPath, args, std::nullopt, redirects);
+    if (result != 0) 
+    {
+        llvm::sys::fs::remove(outputPath);
+        return std::nullopt;
+    }
+
+    auto bufferOrError = llvm::MemoryBuffer::getFile(outputPath);
+    llvm::sys::fs::remove(outputPath);
+    if ( ! bufferOrError) 
+    {
+        return std::nullopt;
+    }
+
+    return bufferOrError.get()->getBuffer().str();
+}
+
+std::vector<std::string> 
+parseIncludePaths(std::string const& compilerOutput) 
+{
+    std::vector<std::string> includePaths;
+    std::istringstream stream(compilerOutput);
+    std::string line;
+    bool capture = false;
+
+    while (std::getline(stream, line)) 
+    {
+        if (line.find("#include <...> search starts here:") != std::string::npos) 
```

Comment:
Is it the same format for all compilers we know about?

Reply:
GCC and Clang has the same format, yes. This method does not work on MSVC, I will add a `#ifdef` to avoid calling this on MSVC. I guess we will not have that "issue" with that compiler.

Reply:
>  I will add a #ifdef

Note that this needs to be a runtime check.

Reply:
Yeah, I had a confusion when I wrote the message here but I did it Ok in the code 👍